registry/regclient/profile.go

```diff
@@ -38,9 +38,44 @@ func (c *Client) CreateProfile(p *registry.Profile, pk crypto.PrivKey) (*registr
 	return c.doJSONProfileReq("POST", pro)
 }
 
-// ProveProfileKey associates a public key with a profile by proving this user
-// can sign messages with the private key
-func (c *Client) ProveProfileKey(p *registry.Profile, pk crypto.PrivKey) (*registry.Profile, error) {
+// ProveKeyForProfile proves to the registry that the user owns the profile and
+// is associating a new keypair
+func (c *Client) ProveKeyForProfile(p *registry.Profile) (map[string]string, error) {
+	if c == nil {
+		return nil, registry.ErrNoRegistry
+	}
+
+	// Ensure all required fields are set
+	if p.ProfileID == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: ProfileID required")
+	}
+	if p.Username == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: Username required")
+	}
+	if p.Email == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: Email required")
+	}
+	if p.Password == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: Password required")
+	}
+	if p.PublicKey == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: PublicKey required")
+	}
+	if p.Signature == "" {
+		return nil, fmt.Errorf("ProveKeyForProfile: Signature required")
+	}
+
+	// Send proof request to the registry
+	res := RegistryResponse{}
+	err := c.doJSONRegistryRequest("PUT", "/registry/provekey", p, &res)
```

Comment:
Can we have this path as a constant somewhere, it will align better with the upcoming API work.